Fix: Make STLLoader work in webworkers.

couchninja opened the pull request:
In a webworker, the window is not initialized, so this used to throw a ReferenceError. This change will prevent that error and enable it to load normally.

couchninja committed:
Fix: Make STLLoader work in webworkers.

In a webworker, the window is not initialized, so this used to throw a ReferenceError. This change will prevent that error and enable it to load normally.

Mugen87 commented:
I think we should change this in all loaders that have this line of code.
```javascript
if ( window.TextDecoder !== undefined )
```
Lffected loaders: `3MFLoader`,`AMFLoader`, `LegacyGLTFLoader`, `GLTFLoader`, `FBXLoader`, `PLYLoader`, `SEA3D`, `STLLoader`.

donmccurdy commented:
> I think we should change this in all loaders that have this line of code.

+1. I assume `TextDecoder` is still available in a WW though, it's just the `window` global that is missing? In that case maybe: 

```js
if ( typeof TextDecoder !== 'undefined' )
```

mrdoob commented:
@donmccurdy LGTM!

couchninja committed:
- extend fix of using file loaders in webworkers to all file workers
- enable file loaders to access TextDecoder in webworkers when window is not defined

couchninja commented:
Thanks for the feedback everyone! Seems like #12654 will fix this problem in a more structural way, so this pull request can be closed in favor of that one.
